model/src/id.rs

```diff
@@ -41,269 +42,645 @@ pub(crate) mod string {
         serializer.collect_str(value)
     }
 
-    pub fn deserialize<'de, T: From<u64>, D: Deserializer<'de>>(
+    pub fn deserialize<'de, T: From<NonZeroU64>, D: Deserializer<'de>>(
         deserializer: D,
     ) -> Result<T, D::Error> {
         deserializer.deserialize_any(IdVisitor(PhantomData))
     }
 }
 
 use serde::{Deserialize, Serialize};
-use std::fmt::{Display, Formatter, Result as FmtResult};
+use std::{
+    fmt::{Display, Formatter, Result as FmtResult},
+    num::NonZeroU64,
+};
+
+/// Unique ID of an application.
+#[allow(clippy::unsafe_derive_deserialize)]
+#[derive(Clone, Copy, Debug, Deserialize, Eq, Hash, Ord, PartialEq, PartialOrd, Serialize)]
+pub struct ApplicationId(#[serde(with = "string")] pub NonZeroU64);
+
+impl ApplicationId {
+    /// Creates a non-zero application ID without checking the value.
```

Comment:
Just the verbiage `Create` is fine, no need to describe it from a third party perspective.